### DISCUSSION TITLE
Methods and attributes list UI

### DISCUSSION THREAD

**Discussion by smonff** [joaotavora/eglot#1445] [Q&A] [answered]
Simple question: does Eglot provide a feature that allows, e.g., for a specific class, to list all it's attributes and methods?

Something [comparable to lsp-ui-imenu](https://emacs-lsp.github.io/lsp-ui/#lsp-ui-imenu) (see screenshot attached). Could be a third-party package.

If not, what paradigm should I use to achieve a comparable feature?

![image](https://github.com/user-attachments/assets/f3539164-85c4-45c9-836a-70b9f4e40d46)

**Comment by joaotavora** [comment, marked as answer]
Not in Eglot core, no.  Eglot provides relatively meager UI enhancements.  I don't know what LSP feature is behind this.  if it's within the LSP spec (and _not_ a server specific interface, of which Rust-analyzer has many) it could make it's way into Eglot (maybe in coordination with a third-party package, maybe not).

Cc: @nemethf

**Reply by smonff** [reply]
Thanks for the answer. The screenshot example is from official [lsp-ui-imenu documentation](https://emacs-lsp.github.io/lsp-ui/#lsp-ui-imenu). I used that with Perl codebases too.

**Reply by joaotavora** [reply]
> The screenshot example is from official [lsp-ui-imenu documentation](https://emacs-lsp.github.io/lsp-ui/#lsp-ui-imenu). I

I don't know what you meant to state by this, but this is irrelevant. I'm looking for official LSP (as in Language Server Protocol) documentation. `lsp-mode` for Emacs is not that thing.

**Reply by smonff** [reply]
Right: lsp-ui are _higher level UI modules of lsp-mode_ and are optional.

**Reply by joaotavora** [reply]
More importantly: that is completely irrelevant here.  Anything `lsp-ui`, `lsp-mode`, `lsp-thingo` is irrelevant to Eglot's development: they're part of a completely different project.  Eglot is concerned with this document: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/ and this document only.
